TypeScript.py

```diff
@@ -760,33 +767,48 @@ def on_selection_modified(self, view):
-            self.mod = True
-            (lastCommand, args, rept) = view.command_history(0)
-#            print("modified " + view.file_name() + " command " + lastCommand + " args " + str(args) + " rept " + str(rept))
-            if info.preChangeSent:
-                # change handled in on_text_command
-                info.clientInfo.changeCount = self.change_count(view)
-                info.preChangeSent = False
-            elif lastCommand == "insert":
-                if (not "\n" in args['characters']) and info.prevSel and (len(info.prevSel) == 1) and (info.prevSel[0].empty()) and (not info.clientInfo.pendingChanges):
+        logger.log.debug ("enter on_modified " + str(view.id()))
+        # it is a special view
+        if view.window() and view.window().active_view().id() != view.id():
+            logger.log.debug ("enter on_modified: special view. started: %s, init_finished: %s" % 
+                   (NavToCommand.started, NavToCommand.ini_finished))
+
+            if NavToCommand.started and NavToCommand.ini_finished:
+                new_content = view.substr(sublime.Region(0, view.size()))
+                sublime.active_window().run_command("nav_to", {'search_text': new_content})
+
+            logger.log.debug ("exit on_modified: special view. started: %s, init_finished: %s" % 
+                   (NavToCommand.started, NavToCommand.ini_finished))
+        # it is a normal view
+        else:
+            info = self.getInfo(view)
+            if info:
+                info.modified = True
+                if cli.ST2():
+                   info.modCount+=1
+                info.lastModChangeCount = self.change_count(view)
+                self.mod = True
+                (lastCommand, args, rept) = view.command_history(0)
+    #            print("modified " + view.file_name() + " command " + lastCommand + " args " + str(args) + " rept " + str(rept))
+                if info.preChangeSent:
+                    # change handled in on_text_command
                     info.clientInfo.changeCount = self.change_count(view)
-                    prevCursor = info.prevSel[0].begin()
-                    cursor = view.sel()[0].begin()
-                    key = view.substr(sublime.Region(prevCursor, cursor))
-                    sendReplaceChangesForRegions(view, staticRegionsToRegions(info.prevSel), key)
-                    # mark change as handled so that on_post_text_command doesn't
-                    # try to handle it
-                    info.changeSent = True
-                else:
-                    # request reload because we have strange insert
-                    info.clientInfo.pendingChanges = True
-            self.setOnIdleTimer(100)
+                    info.preChangeSent = False
+                elif lastCommand == "insert":
+                    if (not "\n" in args['characters']) and info.prevSel and (len(info.prevSel) == 1) and (info.prevSel[0].empty()) and (not info.clientInfo.pendingChanges):
```

Comment:
Turn this into a variable, split it across multiple lines.

Name it something like `should_foo_blah_bleep`